src/components/freelancerJobs/freelancerPage.tsx

```diff
@@ -0,0 +1,213 @@
+						</div>
+						<div>
+							<SkillsButtonsBlock
+								data-toggle="buttons"
+								className="btn-group btn-group-toggle flex-wrap"
+							>
+								{optionButtons}
+							</SkillsButtonsBlock>
+						</div>
+						<CategoryDiv>
+							<CustomSelect
+								options={selection}
+								onChange={choice => setUserChoice(choice as Category)}
+								isClearable
+							/>
+						</CategoryDiv>
+						<div>
+							<SkillsLabel>{`${t('FreelancerPage.price')}`}</SkillsLabel>
+							<P>
+								${slider[0]}-{slider[1]}
+							</P>
+							<Typography id="range-slider" gutterBottom></Typography>
+							<Slider
+								value={slider}
+								onChange={rangeSelector}
+								valueLabelDisplay="auto"
+								min={0}
+								max={10000}
+							/>
+						</div>
+						<div>
+							<Column>
+								<ul>
+									<Li>{`${t('JobPostPage.shortDuration')}`}</Li>
+									<Li>{`${t('JobPostPage.mediumDuration')}`}</Li>
+									<Li>{`${t('JobPostPage.longDuration')}`}</Li>
+								</ul>
+							</Column>
+							<Column>
+								<CheckLabel>
+									{checkList.map((item: string, index: number) => (
+										<div key={index}>
+											<input value={item} type="radio" name="gender" onChange={handleChange} />
+											<Span>{item}</Span>
+										</div>
+									))}
+								</CheckLabel>
+							</Column>
+						</div>
+					</ColumnSmall>
+					<ColumnBig>
+						<SearchInput
+							type="text"
+							placeholder={`${t('FreelancerPage.search')}`}
+							value={search}
+							onChange={e => setSearch(e.target.value)}
+						/>
+						<ClearBtn onClick={ClearFilters}>{`${t('FreelancerPage.clear')}`}</ClearBtn>
+						<ul>
+							{posts
+								.filter((post: IPost) => {
+									if (search === '') {
+										return post;
+									} else if (post.jobTitle.toLowerCase().includes(search.toLowerCase())) {
+										return post;
+									} else if (post.jobDescription.toLowerCase().includes(search.toLowerCase())) {
+										return post;
+									}
+								})
+								.filter((post: IPost) => {
+									const skill = post.jobSkills.map(skill => {
+										return skill.name;
+									});
+									if (
+										post.jobCategory.name === userChoice.label &&
+										slider[0] <= post.fromHourRate &&
+										slider[1] >= post.fromHourRate &&
+										JSON.stringify(radio) === JSON.stringify(post.jobDuration) &&
+										JSON.stringify(skill).includes(JSON.stringify(skills))
+									) {
+										return post;
+									} else if (userChoice.label === '' && skills.length === 0) {
+										return post;
+									}
+								})
+								.map((postData: IPost) => (
+									<Li key={postData.id}>
+										<Link to={`/post-job/${postData.id}`}>
+											<TitleStyled>{postData.jobTitle}</TitleStyled>
+											<DescriptionDataStyled>{postData.jobDescription}</DescriptionDataStyled>
+										</Link>
+									</Li>
+								))}
+						</ul>
+					</ColumnBig>
+				</>
+			)}
+		</>
+	);
+};
```

Comment:
component is too big, move logic to custom hook or/and device to smaller components

Reply:
added smaller components